### ISSUE TITLE
Entity restriction is not available with GLPI Object

### ISSUE THREAD

**Issue by miguelanruizimagu** [pluginsGLPI/formcreator#2342]
**Describe the bug**
To make a question in a form, over the dropdown selection you can choose how to get a restriction in the lists. But when you choose a GLPI Object as the type of question, this selection is not possible.

**To Reproduce**
1. Go to Administration -> Forms.
2. Select whatever form that you need.
3. Create a question with GLPI Object type answer.
4. The Entity restriction option is not shown. --> :(

**Expected behavior**
Currently is not available Entity restriction:
![image](https://user-images.githubusercontent.com/88452851/128227309-4bc7af19-2acc-4ad7-a2f5-e13477839f6d.png)
May be exist possibility to create a restriction with GLPI Objects too.

Yellow highlighted is the expected results:
![image](https://user-images.githubusercontent.com/88452851/128227378-28cddb86-c91a-4789-ba9d-d5c53f008ab1.png)

**Additional context**
The main problem is that you cannot filter dropdowns based on user groups for GLPI Objects over forms location.

**Comment by btry** [comment]
Hi

I found 2 problems to show / hide the entity restriction dropdown. Please test the above fix and feedback.

- open the dialog to create a question
- select GLPI Object and check if the entity restrictin apears
- select Dropdown and see ig the entity restriction is still visible
- select Actors and check the entity restriction disappears

**Comment by miguelanruizimagu** [comment]
No sr, at last mentioned step, the Entity restriction persist.
![image](https://user-images.githubusercontent.com/88452851/128243332-b0a9b6ec-f0fe-4629-863e-ac1190f43a01.png)

You have to reload page and it will dissapear. Then i got to do it manually and forced, changing next line on (How to apply user restriction): 

"/var/www/html/glpi/marketplace/formcreator/inc/field/dropdownfield.class.php"  Line 896;

![image](https://user-images.githubusercontent.com/88452851/128243847-4f472983-7429-42c8-aa16-b612d76c2061.png)

Apparently this workaround works.

**Comment by btry** [comment]
No, your change impacts the default value of the setting. 

If you aplied the patch while the question edition dialog was open and tried with my instruction, then you had bad HTML preventing the setting to hide in  the last step. If I'm right, test again after refreshing the whole HTML page. I'll do also a re-check on my side.

**Comment by btry** [comment]
It works fine for me.

**Comment by miguelanruizimagu** [comment]
> No, your change impacts the default value of the setting.
> 
> If you aplied the patch while the question edition dialog was open and tried with my instruction, then you had bad HTML preventing the setting to hide in the last step. If I'm right, test again after refreshing the whole HTML page. I'll do also a re-check on my side.

No, patch was applied after. 

And patch is a bad way to apply the restriction but it is just temporary, because i need to restrict GLPI Object lists by user. It's the fast way.

**Comment by miguelanruizimagu** [comment]
> Hi
> 
> I found 2 problems to show / hide the entity restriction dropdown. Please test the above fix and feedback.
> 
> * open the dialog to create a question
> * select GLPI Object and check if the entity restrictin apears
> * select Dropdown and see ig the entity restriction is still visible (and appears only once)
> * select Actors and check the entity restriction disappears

I repeated mentioned steps, but i got same results:
![image](https://user-images.githubusercontent.com/88452851/128250224-095b2894-c017-4334-b5f8-1570929446c7.png)

At the end, Entity restriction field, did not disappear.

**Comment by miguelanruizimagu** [comment]
So the thing is right now is not possible apply Entity restriction when type is GLPI Object, and additionally, the field don't disappear.

**Comment by btry** [comment]
Can you check with browser's developer tools if the table row containing the label "Entity restrictrion" has the  class **plugin_formcreator_question_specific** ? JS code uses this class to remove the row from the DOM when you change the question type.

**Comment by miguelanruizimagu** [comment]
Okay, so repeating the steps at 3:
![image](https://user-images.githubusercontent.com/88452851/128254072-f52a77c9-ce83-49d9-a79a-b739f432764d.png)

And last one (clicking Actor):
![image](https://user-images.githubusercontent.com/88452851/128254228-b6f2688c-1728-4bdd-abea-5fd40c067284.png)

**Comment by btry** [comment]
In the 1st screenshot you highlighted the wrong TR tag. 

![image](https://user-images.githubusercontent.com/14139801/128254874-1a0dd2b8-79c8-4d9a-bf14-5ecabe8eaf55.png)

The problem is that the class is not added. Did you apply the patch by using the patch tool ? I see quite often some unexpected issues related people applying manually a patch and breaking a few things.

**Comment by btry** [comment]
Ok for the isntallation of GLPI. In your last screenshot I see the same : you highlighted the wront TR tag. You must see the closest parent TR tag starting from the text Entity restriction.

The one you show is the previous sibling of the closest TR from tne text Entity restriction.

What I need to understand is why it works for me and not for you. I want to ensure that the way you applied the patch is 100% reliable.

**Comment by miguelanruizimagu** [comment]
Im sorry
![image](https://user-images.githubusercontent.com/88452851/128259289-7c5a7e82-4efa-4fcb-9f1a-08fc3581ed34.png)

It should be that u need.

**Comment by btry** [comment]
As you can see the class **plugin_formcreator_question_specific** is missing in the TR classes

![image](https://user-images.githubusercontent.com/14139801/128261040-0e2eac8b-1934-4964-92e1-fd4641fde0f0.png)

The patch should address this problem. How did you applied the patch ? manually, with the patch tool, or with git ?

**Comment by btry** [comment]
Hi

The screenshots shows the same problem I observed **without the patch**. If I understand you well, the patch does not works for you, but I need to check if the patch actually fixes the HTML of the dialog. As you don't provide this information, I cannot help you further. 

As I reproduced the problem you reported and the fix worked at least for me, I merge it.

As you found a solution which satisfy you, I close this issue.

**Comment by miguelanruizimagu** [comment]
I am really sorry, you are right, i did not see the patch before.
![image](https://user-images.githubusercontent.com/88452851/128361505-78e84151-e3d2-40ae-8ff8-7b36d35ea466.png)

Now it is displayed, but this resctriction with GLPI Objects doesnt work fine, should i create a new ticket?

**Comment by btry** [comment]
Good !

Lets continue here.

We need to know if the value is properly saved in DB after you clicked the "save" button of the question.

Please set the non working value in the question, then find in the table glpi_plugin_formcreator_questions the row of the question you saved. Assuming you found its ID, please show the result if the following request. Of course replace 42 with the ID you found.

```sql
SELECT * FROM glpi_plugin_formcreator_questions WHERE ID=42
```

**Comment by miguelanruizimagu** [comment]
Of course, let me show you:
![image](https://user-images.githubusercontent.com/88452851/128381446-fb7cc6c6-257a-4a9a-a714-d8bf41e688a3.png)

Besides this, something else is wrong with "itemtype" over Values field. I'm going to open a new topic because it's different, then i will explain what happen.

**Comment by btry** [comment]
Hi

Indeed, the value is not saved. Here is a fix.

**Comment by btry** [comment]
Hi 

@miguelanruizimagu do you have feedback for the fix #2350 ?

**Comment by miguelanruizimagu** [comment]
Yes sr, im so sorry, that solution works fine.

**Comment by btry** [comment]
Hi,

Thank you for your feedback. The fix is merged.
